backend/core.go

```diff
@@ -72,9 +72,14 @@ type CallResourceResponse struct {
 	Body    []byte
 }
 
+// CallResourceResponseSender used for sending resource call responses.
+type CallResourceResponseSender interface {
+	Send(*CallResourceResponse) error
+}
+
 // CallResourceHandler handles resource calls.
 type CallResourceHandler interface {
-	CallResource(ctx context.Context, req *CallResourceRequest) (*CallResourceResponse, error)
+	CallResource(ctx context.Context, req *CallResourceRequest, sender CallResourceResponseSender) error
```

Comment:
In hindsight, could have kept the original `CallResourceHandler` and created a new `CallResourceStreamHandler` but would add yet another interface for plugin developer to decide which one to implement.

Could maybe rename this to `CallResourceStreamHandler` and create yet another wrapper `CallResourceHandler` that has the same contract as before this change if your only interested in returning one response.

Could also have skipped changing the `CallResourceHandler` alltogether and hidden the streaming support for now.

Reply:
> In hindsight, could have kept the original CallResourceHandler and created a new CallResourceStreamHandler but would add yet another interface for plugin developer to decide which one to implement. 

I think its better that we have fewer contracts to implement. Even if this might be a bit more complex I think mulitiple options would be more confusing. 

> Could also have skipped changing the CallResourceHandler alltogether and hidden the streaming support for now.

I still believe this can be used for websockets and therefore I think this will be used for demos in 7.0